### PR TITLE
[HEVCE] Fix possible DPB overflow

### PR DESCRIPTION
**Pull request by lakulako**
Dpb index was checked against NumRefFrame instead of array size
On ICL where NumRefFrame can be > DPB_SIZE led to assertion fail.

Issue: MDP-45453

### CODE REVIEW

**Comment by lakulako** [comment]
Exploring failed checks, pls don't merge.

File: _studio/mfx_lib/encode_hw/h265/src/mfx_h265_encode_hw_utils.cpp [hunk 1]
```diff
@@ -2830,7 +2830,7 @@ void UpdateDPB(
     MFX_SORT_STRUCT((dpb + st0), mfxU16(end - st0), m_poc, >);
 
     // sliding window over STRs
-    if (end && end == par.mfx.NumRefFrame)
+    if (end && end == MAX_DPB_SIZE)
```
Comment (eistarov) [inline _studio/mfx_lib/encode_hw/h265/src/mfx_h265_encode_hw_utils.cpp]: You're breaking sliding window here.
HEVC encoder DPB should not contain more refs than NumRefFrame.
If NumRefFrame > MAX_DPB_SIZE consider it parameters validation issue.

**Comment by lakulako** [comment]
Updated. Now limits NumRefFrame to maxDPB-1, i.e. dpb without current frame

**Comment by lakulako** [comment]
Shouldn't be merged together with https://github.com/Intel-Media-SDK/MediaSDK/pull/649.
Both fix same issue different ways.

**Pull request merged**